File: backend/internal/types/reviews.go
package types

import (
	"time"

	"github.com/google/uuid"
)

type MovieReview struct {
	ID        uuid.UUID `json:"id"`
	MovieID   uuid.UUID `json:"movie_id"`
	Review    string    `json:"review"`
	Rating    float32   `json:"rating"`
	IsSpoiler bool      `json:"is_spoiler"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
	User      AuthUser  `json:"user"`
}

File: frontend/src/types/reviews.ts
import type { AuthUser } from "./users";

export type Review = {
  id: string; // UUID as string
  movie_id: string; // UUID as string
  review: string;
  rating: number;
  is_spoiler: boolean;
  created_at: string;
  updated_at: string;
  user: AuthUser;
};
